lib/sequent/support/database.rb

```diff
@@ -66,21 +101,27 @@ def self.with_schema_search_path(search_path, db_config, env = ENV['RACK_ENV'])
         fail ArgumentError, 'env is required' unless env
 
         disconnect!
-        original_search_paths = db_config[:schema_search_path].dup
 
         if ActiveRecord::VERSION::MAJOR < 6
           ActiveRecord::Base.configurations[env.to_s] =
             ActiveSupport::HashWithIndifferentAccess.new(db_config).stringify_keys
         end
 
-        db_config[:schema_search_path] = search_path
-
-        ActiveRecord::Base.establish_connection db_config
+        if Sequent.configuration.can_use_multiple_databases
```

Comment:
Afterwards we should set the search_path back to what it was before like the previous version of this method. So in short:

```ruby
def self.can_use_multiple_databases(search_path)
  original_search_paths = get_current_search_path
  set_search_path(search_path)
  yield
ensure
  set_search_path(original_search_paths)
end
```